eodatasets3/prepare/sentinel_sinergise_L1_prepare.py

```diff
@@ -0,0 +1,219 @@
+    }
+
+
+def prepare_and_write(
+    output_yaml_path: Path,
+    ds_path: Path,
+    product_info: Dict,
+    metadata_xml: Dict,
+    format_correctness: Dict,
+) -> Tuple[uuid.UUID, Path]:
+    with DatasetAssembler(
+        metadata_path=output_yaml_path, dataset_location=ds_path
+    ) as p:
+        p.datetime = product_info["timestamp"]
+        p.properties["eo:instrument"] = "MSI"
+        p.properties["eo:platform"] = "sentinel-2a"
+        p.properties["odc:processing_datetime"] = (
+            format_correctness["creation_date"].split("=")[1].replace("T", " ")
+        )
+        p.properties["odc:dataset_version"] = f"1.0.{p.processed:%Y%m%d}"
+        p.properties["odc:producer"] = "sinergise.com"
+        p.properties["odc:product_family"] = "level1"
+        p.properties["eo:sun_elevation"] = metadata_xml["solar_zenith"]
+        p.properties["eo:sun_azimuth"] = metadata_xml["solar_azimuth"]
+        p.properties["eo:gsd"] = metadata_xml["resolution"]
+        p.properties["eo:cloud_cover"] = metadata_xml["cloud"]
+        p.properties["sentinel:sinergise_product_name"] = product_info[
+            "synergise_product_name"
+        ]
+        p.properties["sentinel:sinergise_product_id"] = product_info[
+            "synergise_product_id"
+        ]
+        p.properties["odc:file_format"] = "JPEG2000"
+        p.properties["odc:region_code"] = product_info["region_code"]
+        p.properties["sentinel:data_type"] = "Level-1C"
+        p.properties["sentinel:product_type"] = "S2MSI1C"
+        p.properties["sentinel:software_version"] = format_correctness[
+            "creator_version"
+        ]
+        p.properties["sentinel:source_system"] = format_correctness["source_system"]
+        p.properties["sentinel:datastrip_metadata"] = format_correctness[
+            "datastrip_metadata"
+        ]
+        p.properties["sentinel:downlink_priority"] = metadata_xml["downlink_priority"]
+        p.properties["sentinel:datastrip_id"] = metadata_xml["datastrip_id"]
+
+        directory = [f for f in listdir(ds_path) if isfile(join(ds_path, f))]
+
+        for dataset in directory:
+            if ".jp2" in dataset and "preview" not in dataset and "TCI" not in dataset:
+                band = dataset.replace(".jp2", "").replace("B", "")
+                name = SENTINEL_MSI_BAND_ALIASES[band]
+                path = Path(ds_path) / dataset
+                p.note_measurement(path=path, name=name)
+
+        return p.done()
+
+
+@click.command(help=__doc__)
+@click.option(
+    "--product",
+    type=PathPath(),
+    required=True,
+    help="Path to productInfo.json in sinergise dataset",
+)
+@click.option(
+    "--metadata_xml",
+    type=str,
+    required=True,
+    help="Path to metadata.xml in sinergise dataset",
+)
+@click.option(
+    "--format_correctness",
+    type=str,
+    required=True,
+    help="Path to FORMAT_CORRECTNESS.xml in sinergise dataset",
+)
+@click.option(
+    "--dataset_document",
```

Comment:
We should use dashes as separators in the names to stay consistent with other commands

(Click will still give them to your function with underscores)

```suggestion
    "--metadata-xml",
    type=str,
    required=True,
    help="Path to metadata.xml in sinergise dataset",
)
@click.option(
    "--format-correctness",
    type=str,
    required=True,
    help="Path to FORMAT_CORRECTNESS.xml in sinergise dataset",
)
@click.option(
    "--dataset-document",
```